src/group/managed_group/mod.rs

```diff
@@ -186,54 +425,223 @@ impl ManagedGroup {
         &mut self,
         credential_bundle: &CredentialBundle,
     ) -> Result<Vec<MLSMessage>, ManagedGroupError> {
-        unimplemented!()
+        // Create new KeyPackageBundle for own leaf
+        let tree = self.group.tree();
+        let existing_key_package = tree.get_own_key_package_ref();
+        let key_package_bundle = KeyPackageBundle::from_rekeyed_key_package(existing_key_package);
+        drop(tree);
+
+        // Create UpdateProposal
+        let mut plaintext_messages = vec![self.group.create_update_proposal(
+            &self.aad,
+            credential_bundle,
+            key_package_bundle.get_key_package().clone(),
+        )];
+
+        // Include pending proposals into Commit
+        let mut messages_to_commit = self.pending_proposals.clone();
+        messages_to_commit.extend_from_slice(&plaintext_messages);
+
+        // Create Commit over all proposals
+        let (commit, _welcome_option, kpb_option) =
+            self.group
+                .create_commit(&self.aad, &credential_bundle, &messages_to_commit, false)?;
+
+        // Add the Commit message to the other pending messages
+        plaintext_messages.append(&mut vec![commit]);
+
+        // Take the new KeyPackageBundle and save it for later
+        let kpb = match kpb_option {
+            Some(kpb) => kpb,
+            None => return Err(ManagedGroupError::Unknown),
+        };
+        self.own_kpbs.push(kpb);
+
+        // Convert MLSPlaintext messages to MLSMessage and encrypt them if required by
+        // the configuration
+        let mls_messages = self.plaintext_to_mls_messages(plaintext_messages);
+
+        Ok(mls_messages)
     }
 
     /// Creates a proposal to update the own leaf node
     pub fn propose_self_update(
         &mut self,
         credential_bundle: &CredentialBundle,
     ) -> Result<Vec<MLSMessage>, ManagedGroupError> {
-        unimplemented!()
+        let tree = self.group.tree();
+        let existing_key_package = tree.get_own_key_package_ref();
+        let key_package_bundle = KeyPackageBundle::from_rekeyed_key_package(existing_key_package);
```

Comment:
I'd expect this to be something like `let new_key_package_bundle = existing_key_package_bundle.rekey()`

Reply:
Sure, we could change that. Let's do that in a separate PR though.

Reply:
#235